Support arbitrary blob signing (#283)
notaryproject/notaryproject: specs/signature-envelope-cose.md

CBOR Object Signing and Encryption (COSE). COSE ([RFC8152](https://datatracker.ietf.org/doc/html/rfc8152)) is a CBOR based envelope format for digital signatures over any type of payload (e.g. CBOR, JSON, binary).
Notary Project specifically supports [COSE_Sign1_Tagged](https://datatracker.ietf.org/doc/html/rfc8152#section-4.2) as a signature envelope.

> ## OCI Storage

Reviewer: ```suggestion
## OCI Signature Storage
```

    }
}
```

> ## Detached Signature Storage

Reviewer: Technically oci signatures are detached signatures, can we say `Blob Signature Storage` ?

> ## Detached Blob Signature Storage

Reviewer 2: ```suggestion
## Blob Signature Storage
```

> For detached signatures associated with arbitrary blobs, a COSE signature envelope will be stored on the file system as a binary file with `cose` as the file extension.

Reviewer: ```suggestion
For detached signature associated with blob, a COSE signature envelope will be stored on the file system as a binary file with `cose` as the file extension.
```

COSE detached signature file contains
1. Blob's payload descriptor
1. Signed Attributes
1. Unsigned Attributes

> 1. Signature

Reviewer: This also applies to OCI signature. We should either omit it or make it generic

Author: Removed it

}
```

Example of the Notary Project detached signature payload:

```jsonc
{
  "targetArtifact": {
    "mediaType": "application/octet-stream",
    "digest": "sha256:2f3a23b6373afb134ddcd864be8e037e34a662d090d33ee849471ff73c873345",

>     "size": 1024

Reviewer 3: Do we support `annotation` in the payload? I remember  in this [PR](https://github.com/notaryproject/notation/pull/811/files#diff-30187fc97f8fa76aa528babf8e93d521e4b412bbe47dea781678c6d4d96088e1) , 'notation blob sign --user-metadatra` will add annotation to the signature blob

}
```

> Example of the Notary Project blob signature payload:

Reviewer: The payload format is identical for both blobs and OCI. In my opinion, we don't require separate examples for blobs and OCI.

> Example of the Notary Project detached signature payload:

Reviewer: Since we are not changing signature envelope, IMO we dont need new example.

> Example of the Notary Project detached signature payload:

Reviewer: ```suggestion
Example of the Notary Project blob signature payload:
```
